admin/test/test_release.py

```diff
@@ -661,3 +667,637 @@ def test_publish_non_release_fails(self):
+            },
+        )
+
+        repo_contents = {
+            'results/omnibus/0.3.3/fedora-20/clusterhq-flocker-cli-0.3.3.noarch.rpm': '',  # noqa
+            'results/omnibus/0.3.3/fedora-20/clusterhq-flocker-node-0.3.3.noarch.rpm': '',  # noqa
+            'results/omnibus/0.3.3/centos-7/clusterhq-flocker-cli-0.3.3.noarch.rpm': '',  # noqa
+            'results/omnibus/0.3.3/centos-7/clusterhq-flocker-node-0.3.3.noarch.rpm': '',  # noqa
+
+        }
+
+        self.upload_rpms(
+            aws=aws,
+            yum=FakeYum(),
+            scratch_directory=self.scratch_directory,
+            target_bucket=self.target_bucket,
+            version='0.3.3',
+            build_server=self.create_fake_repository(files=repo_contents),
+        )
+
+        expected_files = set()
+        for operating_system in self.operating_systems:
+            for file in [
+                'clusterhq-flocker-cli-0.3.3.noarch.rpm',
+                'clusterhq-flocker-node-0.3.3.noarch.rpm',
+                'repodata/repomd.xml',
+            ]:
+                path = os.path.join(
+                    'marketing',
+                    operating_system['distro'],
+                    operating_system['version'],
+                    operating_system['arch'],
+                    file,
+                )
+                expected_files.add(path)
+
+        files_on_s3 = aws.s3_buckets[self.target_bucket].keys()
+
+        self.assertTrue(expected_files.issubset(set(files_on_s3)))
+
+    @skipUnless(which('yum'), "Tests require the ``yum`` command.")
+    @skipUnless(which('createrepo'),
+        "Tests require the ``createrepo`` command.")
+    @skipUnless(which('yumdownloader'),
+        "Tests require the ``yumdownloader`` command.")
+    def test_real_yum_utils(self):
+        """
+        Calling :func:`update_repo` with real yum utilities creates a
+        repository in S3.
+        """
+        source_repo = FilePath(tempfile.mkdtemp())
+        self.addCleanup(source_repo.remove)
+
+        FilePath(__file__).sibling('test-repo').copyTo(source_repo)
+        repo_uri = 'file://' + source_repo.path
+
+        aws = FakeAWS(
+            routing_rules={},
+            s3_buckets={
+                self.target_bucket: {},
+            },
+        )
+
+        class RealYum(object):
+            def get_dispatcher(self):
+                return yum_dispatcher
+
+        self.update_repo(
+            aws=aws,
+            yum=RealYum(),
+            rpm_directory=self.rpm_directory,
+            target_bucket=self.target_bucket,
+            target_key=self.target_key,
+            source_repo=repo_uri,
+            packages=self.packages,
+        )
+
+        files = [
+            'clusterhq-flocker-cli-0.3.3-0.dev.7.noarch.rpm',
+            'clusterhq-flocker-node-0.3.3-0.dev.7.noarch.rpm',
+            'repodata/1988e623dfb7204ec981d2a2ab1a38da6c0f742717d184088dd0a0f344f6e89c-filelists.sqlite.bz2',  # noqa
```

Comment:
I guess the fact that these files match those in the test-repo is just because the repos are identical, rather than that they get copied from one to another. Perhaps a comment would be useful here.